packages/twenty-server/src/engine/core-modules/auth/guards/saml-auth.guard.ts

```diff
@@ -9,33 +9,44 @@ import {
 } from 'src/engine/core-modules/auth/auth.exception';
 import { SamlAuthStrategy } from 'src/engine/core-modules/auth/strategies/saml.auth.strategy';
 import { SSOService } from 'src/engine/core-modules/sso/services/sso.service';
+import { GuardErrorManagerService } from 'src/engine/core-modules/guard-manager/services/guard-error-manager.service';
 
 @Injectable()
 export class SAMLAuthGuard extends AuthGuard('saml') {
-  constructor(private readonly sSOService: SSOService) {
+  constructor(
+    private readonly sSOService: SSOService,
+    private readonly guardErrorManagerService: GuardErrorManagerService,
+  ) {
     super();
   }
 
   async canActivate(context: ExecutionContext) {
+    const request = context.switchToHttp().getRequest();
+
+    let identityProvider: Awaited<
```

Comment:
new pattern: Awaited<>. Can we instead introduce a type and export it (and use Promise<type>)